### PR TITLE
Added byday recurrence feature

### PR DESCRIPTION
**Pull request by Julian1998**
Instructions:

1. create a new event
2. click on more...
3. go to repeating
4. create a new recurrence with byday options

Possible scenarios:

1. creating a new event with byday options
2. modifying the current byday options

All events should render correctly even when changing the first day of the week or/and language

All scenarios work fine in Chrome

### CODE REVIEW

File: js/app/controllers/recurrencecontroller.js [hunk 1]
```diff
@@ -86,10 +109,34 @@ app.controller('RecurrenceController', function($scope) {
...
+	$scope.$watch('byDay', function (newValue) {
+		if(!ctrl.loading) {
+			ctrl.transferDaysToByDay(newValue);
+		}
+	}, true);
 
-});
+	ctrl.transferDaysToByDay = function(newDays) {
+		angular.forEach(newDays, function(value, key) {
+			if(typeof $scope.properties.rrule.byday === 'undefined') {
```
Comment (DeepDiver1975) [inline js/app/controllers/recurrencecontroller.js]: angular.isUndefined

File: js/app/controllers/recurrencecontroller.js [hunk 2]
```diff
@@ -64,7 +64,7 @@ app.controller('RecurrenceController', function($scope) {
 				$scope.rruleNotSupported = true;
 			}
 
-			if (typeof $scope.properties.rrule.parameters !== 'undefined') {
+			if (!angular.isUndefined($scope.properties.rrule.parameters)) {
```
Comment (DeepDiver1975) [inline js/app/controllers/recurrencecontroller.js]: dont use inverted logic - https://docs.angularjs.org/api/ng/function/angular.isDefined

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

**Comment by DeepDiver1975** [comment]
bildschirmfoto von 2018-05-30 19-15-35

jslint hates you ;-)

**Comment by DeepDiver1975** [comment]
The day picker is not working for me ....

**Comment by Julian1998** [comment]
jslint didn't tell me... alright I will have a look at it... Which day picker exactly?

**Comment by DeepDiver1975** [comment]
> Which day picker exactly?

the monday, tueseday and so on one .... will show you these days live ...

**Comment by Julian1998** [comment]
Seems like jslint didn't build the js sources correctly because of this error. I will fix it and then it should build

**Comment by DeepDiver1975** [comment]
@Julian1998 ping

**Comment by DeepDiver1975** [comment]
looks like tests are failing 

```
Firefox 60.0.0 (Linux 0.0.0) The SimpleEvent factory should add the repeating information FAILED
	TypeError: newSimpleData.rrule.parameters is undefined in js/app/models/simpleEventModel.js (line 9)
	repeating@js/app/models/simpleEventModel.js:9:32091
	SimpleEvent/iface.patch@js/app/models/simpleEventModel.js:9:34974
	@tests/js/unit/models/simpleEventModelSpec.js:2156:3
	<Jasmine>
Firefox 60.0.0 (Linux 0.0.0): Executed 161 of 377 (1 FAILED) (0 secs / 0 secs)
 (1 FAILED) (0 secs / 0 secs)

**Review by DeepDiver1975 — APPROVED** [review]

**Pull request merged**